### ISSUE TITLE
authorize_exam_runs task fails every hour

### ISSUE THREAD

**Issue by pdpinch** [mitodl/micromasters#4433]
@sentry-io[bot] commented on [Sun Oct 20 2019](https://github.com/mitodl/bootcamp-ecommerce/issues/318)

Sentry Issue: [MICROMASTERS-333](https://sentry.io/organizations/mit-office-of-digital-learning/issues/1143927971/?referrer=github_integration)

```
OperationalError: cursor "_django_curs_140142796867328_1" does not exist

  File "django/db/utils.py", line 96, in inner
    return func(*args, **kwargs)

OperationalError: cursor "_django_curs_140142796867328_1" does not exist

  File "django/db/models/sql/compiler.py", line 1466, in cursor_iter
    for rows in iter((lambda: cursor.fetchmany(itersize)), sentinel):
  File "django/db/models/sql/compiler.py", line 1466, in <lambda>
    for rows in iter((lambda: cursor.fetchmany(itersize)), sentinel):
  File "django/db/utils.py", line 96, in inner
    return func(*args, **kwargs)
  File "django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "django/db/utils.py", line 96, in inner
    return func(*args, **kwargs)

OperationalError: cursor "_django_curs_140142796867328_1" does not exist

(2 additional frame(s) were not displayed)
...
  File "exams/tasks.py", line 222, in authorize_exam_runs
    api.bulk_authorize_for_exam_run(exam_run)
  File "exams/api.py", line 169, in bulk_authorize_for_exam_run
    program=exam_run.course.program
  File "django/db/models/query.py", line 332, in _iterator
    yield from self._iterable_class(self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size)
  File "django/db/models/query.py", line 63, in __iter__
    for row in compiler.results_iter(results):
  File "django/db/models/sql/compiler.py", line 1469, in cursor_iter
    cursor.close()

OperationalError: cursor "_django_curs_140142796867328_1" does not exist

```

**Comment by pdpinch** [comment]
Usually when I see this error, it's on an admin page, or maybe a CMS page. I just reload the page and then it works. 

In this case, this is a celery task that needs to run periodically. 

Based on the stacktrace in sentry, it seems like this is the problematic code:

https://github.com/mitodl/micromasters/blame/51fdf465d75db880ab7b6ab86df7394128d1c278/exams/api.py#L161-L176

Would it help to use a different query with fewer results? Right now it looks like it iterates through every user in the program, when it could be narrowed to every user who in enrolled the in the course.

**Comment by pdpinch** [comment]
This was fixed by Matt in #4481
